modules/markup/mdstripper/mdstripper.go

```diff
@@ -91,8 +91,7 @@ func (r *stripRenderer) processAutoLink(w io.Writer, link []byte) {
 	}
 
 	// Note: we're not attempting to match the URL scheme (http/https)
-	host := strings.ToLower(u.Host)
-	if host != "" && host != strings.ToLower(r.localhost.Host) {
+	if u.Host != "" && !strings.EqualFold(u.Host, r.localhost.Host) {
```

Comment:
Host should be ASCII only

Reply:
No, backend is ASCII-only, see the Punycode

> In the Domain Name System, these domains use an [ASCII](https://en.wikipedia.org/wiki/ASCII) representation consisting of the prefix "xn--" followed by the [Punycode](https://en.wikipedia.org/wiki/Punycode) translation of the [Unicode](https://en.wikipedia.org/wiki/Unicode) representation of the language-specific alphabet or script glyphs. For example, the [Cyrillic](https://en.wikipedia.org/wiki/Cyrillic) name of Russia's IDN ccTLD is "рф". In Punycode representation, this is "p1ai", and its DNS name is "xn--p1ai".

Reply:
Are you certain that domains will enter in their punycode form into this function? I am not and therefore it's better to use unicode-aware functions.

Reply:
> Are you certain that domains will enter in their punycode form into this function?

![image](https://github.com/user-attachments/assets/b894a58d-c602-435d-8b54-9ecbc4b5dcd8)